src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/MemberAndDeclarationInfo.cs

```diff
@@ -246,7 +246,28 @@ private sealed class MemberNameComparer : IComparer<MemberAndDeclarationInfo>
             public int Compare(MemberAndDeclarationInfo x, MemberAndDeclarationInfo y)
             {
                 var comp = string.Compare(x.Name, y.Name, StringComparison.Ordinal);
-                return comp != 0 ? comp : (y._inheritanceLevel - x._inheritanceLevel);
+                if (comp == 0)
+                {
+                    return y._inheritanceLevel - x._inheritanceLevel;
+                }
+
+                // Field members come first
+                var xIsField = x.MemberType == MemberTypes.Field;
+                var yIsField = y.MemberType == MemberTypes.Field;
+                if (xIsField && yIsField)
+                {
+                    return comp;
+                }
+                if (xIsField)
+                {
+                    return -1;
+                }
+                if (yIsField)
+                {
+                    return 1;
+                }
+
+                return comp;
```

Comment:
It looks like this could be reduced to `return yIsField.CompareTo(xIsField);`

Reply:
I'm not seeing it. When neither is a field, we still want to use `comp` to order.

---
In reply to: [357301187](https://github.com/dotnet/roslyn/pull/40210#discussion_r357301187) [](ancestors = 357301187)

Reply:
Ah, I thought `comp == 0` at this point rather than `comp != 0`. In that case, it could be reduced to the following but that seems less important:
```
var fieldComp = yIsField.CompareTo(xIsField);
return fieldComp == 0 ? comp : fieldComp;
```

---
In reply to: [357307510](https://github.com/dotnet/roslyn/pull/40210#discussion_r357307510) [](ancestors = 357307510,357301187)